### DISCUSSION TITLE
database containerization [Apr 16]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#1848] [Discord Threads]
> 🙋‍♂️ **Acose**

**Comment by o1lab** [comment]
> 🙋‍♂️ **yallouche**  
#### Only the service you reverse proxy will be accessible outside of the docker virtual network, so you database will remain safe.   
####    
#### Even if you wanted to, I don't think reverse-proxy your database would be achievable, it's more than a simple HTTP connection which is what it usually focuses on  
#  
> 🙋‍♂️ **Acose**  
#### Thank you for your consideration! That’s what I have gathered so far. Do you happen to know if this is the usual method a small business organization would be structured if they are self hosting. I am very knew but ambitious in learning all this development processes. For example, having an internet facing admin server to service the database API? Or is there some other way to give access to the database for a fully published app like on the AppStores. I have no idea how to frame this question to google a concrete answer…  
#  
> 💁 **o1lab**  
#### see if this is helpful : one of our user had used it for TrueNAS  
#  
> 💁 **o1lab**  
#### https://github.com/nocodb/nocodb/tree/develop/scripts/setups/TrueNAS  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/964234598185840660)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
